set.mm

```diff
@@ -501775,6 +501788,75 @@ have become an indirect lemma of the theorem in question (i.e. a lemma
       WKWKWKWLCWRIXOXHCBWMEWNCWRFGHWOWPWQ $.
   $}
 
+$(
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+  Acyclic graphs
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+$)
+
+  $c AcyclicGraph $.
+
+  $( Extend class notation with acyclic graphs. $)
+  cacycgr $a class AcyclicGraph $.
+
+  ${
+    $d g f p $.
+    $( Define the class of all acyclic graphs.  A graph is called _acyclic_ if
+       it has no (non-trivial) cycles.  (Contributed by BTernaryTau,
+       11-Oct-2023.) $)
+    df-acycgr $a |- AcyclicGraph =
+              { g | -. E. f E. p ( f ( Cycles ` g ) p /\ f =/= (/) ) } $.
+  $}
+
+  ${
+    $d f g p $.
+    $( An alternate definition of the class of all acyclic graphs that requires
+       all cycles to be trivial.  (Contributed by BTernaryTau, 11-Oct-2023.) $)
+    dfacycgr1 $p |- AcyclicGraph =
+              { g | A. f A. p ( f ( Cycles ` g ) p -> f = (/) ) } $=
+      ( cacycgr cv ccycls cfv wbr c0 wne wa wex cab wceq wal df-acycgr 2exanali
+      wn wi df-ne anbi2i 2exbii xchnxbir abbii eqtri ) DAEZCEBEFGHZUFIJZKZCLALZ
+      RZBMUGUFINZSCOAOZBMABCPUKUMBUGULRZKZCLALUMUJUGULACQUIUOACUHUNUGUFITUAUBUC
+      UDUE $.
+  $}
+
+  ${
+    $d f g G p $.
+    $( The property of being an acyclic graph.  (Contributed by BTernaryTau,
+       11-Oct-2023.) $)
+    isacycgr $p |- ( G e. W -> ( G e. AcyclicGraph <->
+              -. E. f E. p ( f ( Cycles ` G ) p /\ f =/= (/) ) ) ) $=
+      ( vg cv ccycls cfv wbr c0 wne wa wex wn cacycgr wceq fveq2 anbi1d 2exbidv
+      breqd notbid df-acycgr elab2g ) AFZDFZEFZGHZIZUDJKZLZDMAMZNUDUEBGHZIZUILZ
+      DMAMZNEBOCUFBPZUKUOUPUJUNADUPUHUMUIUPUGULUDUEUFBGQTRSUAAEDUBUC $.
+  $}
+
+  ${
+    $d f g G p $.
+    $( The property of being an acyclic graph.  (Contributed by BTernaryTau,
+       11-Oct-2023.) $)
+    isacycgr1 $p |- ( G e. W -> ( G e. AcyclicGraph <->
+              A. f A. p ( f ( Cycles ` G ) p -> f = (/) ) ) ) $=
+      ( vg cv ccycls cfv wbr c0 wceq wal cacycgr fveq2 imbi1d 2albidv dfacycgr1
+      wi breqd elab2g ) AFZDFZEFZGHZIZUAJKZRZDLALUAUBBGHZIZUFRZDLALEBMCUCBKZUGU
+      JADUKUEUIUFUKUDUHUAUBUCBGNSOPAEDQT $.
+  $}
+
+  ${
+    $d P f p $.  $d f F p $.  $d f G p $.
+    $( Any cycle in an acyclic graph is trivial (i.e. has one vertex and no
+       edges).  (Contributed by BTernaryTau, 11-Oct-2023.) $)
+    acycgrcycl $p |- ( G e. AcyclicGraph ->
+              ( F ( Cycles ` G ) P -> F = (/) ) ) $=
+      ( vf vp cacycgr wcel ccycls cfv wbr c0 wceq wi wa cv cvv cwlks w3a adantl
+      wal cycliswlk syl simp2d simp3d breq1 eqeq1 imbi12d breq2 imbi1d sylan9bb
+      wlkv isacycgr1 ibi 19.21bbi adantr vtocl2d ex pm2.43d ) CFGZBACHIZJZBKLZU
+      SVAVAVBMZUSVANDOZEOZUTJZVDKLZMZVCDEBAPPVABPGZUSVACPGZVIAPGZVABACQIJVJVIVK
+      RABCUAABCUKUBZUCSVAVKUSVAVJVIVKVLUDSVDBLZVHBVEUTJZVBMVEALZVCVMVFVNVGVBVDB
+      VEUTUEVDBKUFUGVOVNVAVBVEABUTUHUIUJUSVHVAUSVHDEUSVHETDTDCFEULUMUNUOUPUQUR
+      $.
+  $}
```

Comment:
In general I prefer statements like
```
|- ( ( G e. AcyclicGraph /\ F ( Cycles ` G ) P ) -> F = (/) ) )
```
Rather than the double implication. I believe we have more theorems that can make use of this form, even if it's just an `~ imp` away.

Reply:
Done.